semgrep/semgrep/output.py

```diff
@@ -399,7 +409,10 @@ def build_output(self, color_output: bool) -> str:
         elif output_format == OutputFormat.SARIF:
             return build_sarif_output(self.rule_matches, self.rules)
         elif output_format == OutputFormat.TEXT:
-            return "\n".join(build_normal_output(self.rule_matches, color_output))
+            return "\n".join(
+                list(build_normal_output(self.rule_matches, color_output))
+                + ([self.stats_line] if self.stats_line else [])
```

Comment:
Some users expect no output to be printed when no matches are found. IMO a stats line should be under --verbose and output should only ever be findings.

Reply:
Even in an interactive session @brendongo? I agree if run under a headless shell but for interactive it makes more sense that it tells you how many files it ran on and that there were 0 findings. I get that that differs from grep, but I'm trying to think about the non-poweruser UX.